### PR TITLE
[FEAT]: Migrate Meshery Settings to Functional Component with E2E test

### PR DESCRIPTION
**Pull request by senthil-athiban**
**Notes for Reviewers**

This PR fixes #

- [x] Migrate Meshery Settings from class to functional component
- [x] E2E Test: Behaviour check of Meshery Setting, whether it navigates to selected tab.

**Note for Reviewers**: Followed [Playwright best practices](https://playwright.dev/docs/best-practices#best-practices) by using locators for querying the DOM.

## Screen Recorder

https://github.com/user-attachments/assets/b1578f8b-4f11-4134-ac95-a84880921356

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: ui/tests/e2e/settings.spec.js [hunk 1]
```diff
@@ -50,6 +50,44 @@ test.describe('Settings Page Tests', () => {
...
+    // Navigate to "Prometheus" sub-tab
+    await page.getByRole('tab', { name: 'Prometheus' }).click();
+
+    // Prometheus Base URL subt-tab are displayed
+    await expect(page.locator('label').filter({ hasText: 'Prometheus Base URL' })).toBeVisible();
+
+    // Navigate to "Registry" tab
+    await page.getByRole('tab', { name: 'Registry' }).click();
+
+    // "Mesh Model" component is displayed
```
Comment (leecalcote) [inline ui/tests/e2e/settings.spec.js:79]: Please remove "Mesh".

Please spell "Database" as "Database".

File: ui/tests/e2e/settings.spec.js [hunk 2]
```diff
@@ -50,6 +50,44 @@ test.describe('Settings Page Tests', () => {
...
+    await expect(page.locator('label').filter({ hasText: 'Grafana Base URL' })).toBeVisible();
+
+    // Navigate to "Prometheus" sub-tab
+    await page.getByRole('tab', { name: 'Prometheus' }).click();
+
+    // Prometheus Base URL subt-tab are displayed
+    await expect(page.locator('label').filter({ hasText: 'Prometheus Base URL' })).toBeVisible();
+
+    // Navigate to "Registry" tab
+    await page.getByRole('tab', { name: 'Registry' }).click();
```
Comment (leecalcote) [inline ui/tests/e2e/settings.spec.js:77]: Can we use an id or other static identifier to locate elements other than text that may well change and break the test?

**Review by leecalcote — CHANGES_REQUESTED** [review]

File: ui/tests/e2e/settings.spec.js (hunk 2 above)
Comment (senthil-athiban) [reply ui/tests/e2e/settings.spec.js:77]: Based on the [priority given by Playwright](https://playwright.dev/docs/locators#quick-guide), we can use `getByRole` for querying the DOM. 
And we can use [`getByTestId`](https://playwright.dev/docs/locators#locate-by-test-id) in scenarios where we cannot locate elements by [role](https://playwright.dev/docs/locators#locate-by-role) or [text](https://playwright.dev/docs/locators#locate-by-text).
Additionally, Playwright mentions that we can follow this approach in their [Best Practices section](https://playwright.dev/docs/best-practices#best-practices)

Comment (leecalcote) [reply ui/tests/e2e/settings.spec.js:77]: Thanks! Regex isn't the solution that we're looking for. It suffers from the same fragility. We're looking for a unique identifier to use as the locator, @senthil-athiban

Comment (senthil-athiban) [reply ui/tests/e2e/settings.spec.js:77]: I'll go with unique with the approach of using id as selector.

**Comment by leecalcote** [comment]
@senthil-athiban merge conflict.

@jerensl will you give this a review?

**Comment by leecalcote** [comment]
@senthil-athiban merge conflict.

@jerensl will you give this a review?

**Pull request closed without merging**